supervisor/store/__init__.py

```diff
@@ -194,7 +198,6 @@ async def remove_repository(self, repository: Repository, *, persist: bool = Tru
             await self.data.update()
             self._read_addons()
 
-    @Job(conditions=[JobCondition.INTERNET_SYSTEM])
```

Comment:
I realize this change is a bit strange since @pvizeli you suggested adding a `SUPERVISOR_UPDATED` condition here and instead I removed one. I realized a couple things:
1. We only need to add the condition to `add_repository` as this calls that for adding repositories
1. Removing repositories (which this can do too) doesn't require internet or supervisor to be updated
1. We use this method when restoring backups and during startup and we don't really want to any more checks then strictly necessary during those times